ui/main/src/app/modules/monitoring/components/monitoring-table/monitoring-table.component.ts

```diff
@@ -1,25 +1,80 @@
-import {Component, Input} from '@angular/core';
-import {LineOfMonitoringResult} from '@ofModel/line-of-monitoring-result.model';
-import {TimeService} from '@ofServices/time.service';
-import {Moment} from 'moment-timezone';
+import { Component, Input, OnInit, OnDestroy } from '@angular/core';
+import { LineOfMonitoringResult } from '@ofModel/line-of-monitoring-result.model';
+import { TimeService } from '@ofServices/time.service';
+import { Moment } from 'moment-timezone';
+import { TranslateService } from '@ngx-translate/core';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
 
 @Component({
     selector: 'of-monitoring-table',
     templateUrl: './monitoring-table.component.html',
     styleUrls: ['./monitoring-table.component.scss']
 })
-export class MonitoringTableComponent {
+export class MonitoringTableComponent implements OnInit, OnDestroy {
 
     @Input() result: LineOfMonitoringResult[];
+    exportMonitoringData: Array<any> = [];
+    unsubscribe$: Subject<void> = new Subject<void>();
 
-    constructor(readonly timeService: TimeService) {
+
+    constructor(readonly timeService: TimeService, private translate: TranslateService) {
     }
 
+
     displayTime(moment: Moment) {
 
         if (!!moment) {
             return this.timeService.formatDateTime(moment);
         }
         return '';
     }
+    ngOnInit() {
+        this.initExportMonitoringData();
+
+    }
+
+    initExportMonitoringData(): void {
+
+        let time: string, businessPeriod: string, processName: any, title: any, summary: any, status: any;
+
+        Array.prototype.forEach.call(this.result, (line: LineOfMonitoringResult) => {
```

Comment:
`Array.prototype.forEach.call(this.result, ...`

replace with :
`this.result.forEach( (line: LineOfMonitoringResult) => ...
`

Reply:
ok

Reply:
ok